src/app/components/elements/panels/SelectedEventDetails.tsx

```diff
@@ -1,61 +1,112 @@
+} from "../../../state";
+import { Link } from "react-router-dom";
+import { DateString } from "../DateString";
+import { NOT_FOUND, zeroOrLess } from "../../../services";
+
+export const SelectedEventDetails = ({ eventId }: { eventId: string }) => {
+  const dispatch = useAppDispatch();
+
+  const [studentCount, setStudentCount] = useState(0);
+  const [teacherCount, setTeacherCount] = useState(0);
+
+  useEffect(() => {
+    dispatch(getEvent(eventId));
+  }, [dispatch, eventId]);
+
+  const selectedEvent = useAppSelector((state: AppState) => {
+    return state && state.currentEvent;
+  });
+  const eventBookings = useAppSelector(
+    (state: AppState) => (state && state.eventBookings) || []
+  );
+
+  useEffect(() => {
+    console.log("running rolesOfBookedUsers");
+    let newStudentCount = 0;
+    let newTeacherCount = 0;
+
+    eventBookings.forEach((booking) => {
+      const role = booking.userBooked?.role;
+      if (role === "STUDENT") {
+        newStudentCount++;
+      } else if (role === "TEACHER") {
+        newTeacherCount++;
+      }
+    });
+
+    setStudentCount(newStudentCount);
+    setTeacherCount(newTeacherCount);
+  }, [eventBookings]);
+
+  return (
+    <RS.Card>
+      <RS.CardBody>
+        <h3 className="h-subtitle mb-1">Selected event details</h3>
+        {selectedEvent && selectedEvent !== NOT_FOUND && (
+          <p className="m-0">
+            <strong>Event: </strong>
+            <Link to={`/events/${selectedEvent.id}`} target="_blank">
+              {selectedEvent.title} {selectedEvent.subtitle}
+            </Link>
+            <br />
+            {selectedEvent.location &&
+              selectedEvent.location.address &&
+              selectedEvent.location.address.addressLine1 && (
+                <React.Fragment>
+                  <strong>Location: </strong>
+                  {selectedEvent.isVirtual
+                    ? "Online"
+                    : `${selectedEvent.location.address.addressLine1}, ${selectedEvent.location.address.town}, ${selectedEvent.location.address.postalCode}`}
+                  <br />
+                </React.Fragment>
+              )}
+            <strong>Event status: </strong>
+            <span
+              className={
+                selectedEvent.isCancelled ? "text-danger font-weight-bold" : ""
+              }
+            >
+              {selectedEvent.eventStatus}
+            </span>
+            <br />
+            <strong>Event start: </strong>
+            <DateString>{selectedEvent.date}</DateString> -{" "}
+            <DateString>{selectedEvent.endDate}</DateString>
+            <br />
+            <strong>Booking deadline: </strong>
+            <DateString>{selectedEvent.bookingDeadline}</DateString>
+            <br />
+            <strong>Prepwork deadline: </strong>
+            <DateString>{selectedEvent.prepWorkDeadline}</DateString>
+            <br />
+            {/* Group token is currently JSON Ignored by the API */}
```

Comment:
Is this commented out code really useful? I would remove it unless we're planning to implement this soon.

Reply:
I know you're not keen on comments, but since we haven't fully explored some areas of functionality these are the types of comments I've been leaving in - I feel like it may help us understand in the future if we come to do anything with the group tokens; so far we haven't needed to.